contrib/IECoreUSD/test/IECoreUSD/USDSceneTest.py

```diff
@@ -1878,5 +1878,129 @@ def testDisplayColor( self ) :
 			vertexPlane
 		)
 
+	def testCanReferenceTags( self ) :
+
+		# Write "tags.usda" via SceneInterface.
+
+		tagsFileName = os.path.join( self.temporaryDirectory(), "tags.usda" )
+		root = IECoreScene.SceneInterface.create( tagsFileName, IECore.IndexedIO.OpenMode.Write )
+		child = root.createChild( "child" )
+		child.writeTags( [ "tagA", "tagB" ] )
+		grandChild = child.createChild( "grandChild" )
+		grandChild.writeTags( [ "tagB", "tagC" ] )
+		del root, child, grandChild
+
+		# Write "test.usda" via USD API, referencing "tags.usda".
+
+		fileName = os.path.join( self.temporaryDirectory(), "test.usda" )
+		root = pxr.Usd.Stage.CreateNew( fileName )
+		root.OverridePrim( "/child" ).GetReferences().AddReference( tagsFileName, "/child" )
+		root.GetRootLayer().Save()
+		del root
+
+		# Load "test.usda" via SceneInterface, and check that we can read the tags.
+
+		root = IECoreScene.SceneInterface.create( fileName, IECore.IndexedIO.OpenMode.Read )
+		child = root.child( "child" )
+		self.assertEqual( { str( x ) for x in child.readTags() }, { "tagA", "tagB" } )
```

Comment:
Are you asserting as sets because tagB is listed twice or because the order is indeterminate? I think the former shouldn't be allowed, so might be better to assert as lists if we can do.

Reply:
Good point - it was the latter, but it meant the former could have gone undetected. I've improved this in 6120b9f.